components/appmgt/org.wso2.carbon.appmgt.api/src/main/java/org/wso2/carbon/appmgt/api/APIProvider.java

```diff
@@ -477,4 +476,18 @@ public boolean deleteApp(APIIdentifier identifier, SSOProvider ssoProvider) thro
      */
     public Map<Map<String, String>, Long> getSubscriptionCountByAPPs(String provider, String fromDate, String toDate)
             throws AppManagementException;
+
+
+    /**
+     * Get Application wise policy group list
+     *
+     * @param appId : Application Id
+     * @return list of policy groups
+     * @throws AppManagementException on error
+     */
+    public List<EntitlementPolicyGroup> getPolicyGroupListByApplication(Integer appId) throws
```

Comment:
No need to use the Integer object type as the parameter, just int is enough. Java provides unboxing.